lib/resilient/key.rb

```diff
@@ -3,7 +3,16 @@ class Key
     attr_reader :name
 
     def initialize(name)
-      @name = name
+      @name = name.to_s
+    end
+
+    def ==(other)
+      self.class == other.class && name == other.name
+    end
+    alias_method :eql?, :==
```

Comment:
eql? is what is really needed, but I tend to always define both for good measure.